src/many-sources/abstract-pool.js

```diff
@@ -34,8 +35,15 @@ inherit(AbstractPool, Stream, {
       if (this._queueLim === -1 || this._queue.length < this._queueLim) {
         this._addToQueue(toObs(obj));
       } else if (this._drop === 'old') {
-        this._removeOldest();
-        this._add(obj, toObs);
+        if (this._overlapping) {
+          this._addToCur(toObs(obj));
+          while (this._curSources.length > this._concurLim) {
+            this._removeOldest();
+          }
```

Comment:
I think we should just remove one. Yes, there is a recursion on the other side, but it should always do only one iteration because:
1. We always expect this condition to be true: `this._curSources.length <= this._concurLim`.
2. When we get to this branch of the code `this._curSources.length === this._concurLim` should be true.
3. We remove one from `this._curSources`, so now `this._curSources.length === this._concurLim - 1`.
4. At the following `this._add()` call, we fall into branch at line 33. And that is the end.

But more importantly, with the condition you added we might actually remove none of the oldest. After `this._addToCur(toObs(obj))` we not necessarily have an observable added to `_curSources` because if `toObs(obj)` already anded or ends immediately after we subscribe to it, we don't add it to the `_curSources` (take a look at `_addToCur`). Btw, we need a test case for this.